File: app/src/main/java/com/example/sfmapp/WelcomeActivity.java
package com.example.sfmapp;
import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
public class WelcomeActivity extends AppCompatActivity {

    Button buttonGoReg;
    Button buttonGoSign;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        buttonGoReg =findViewById(R.id.buttonGoReg);
        buttonGoSign =findViewById(R.id.buttonGoSign);

        buttonGoReg.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(WelcomeActivity.this, RegisterActivity.class);
                startActivity(intent);
                overridePendingTransition( android.R.anim.slide_in_left,android.R.anim.slide_out_right);
                WelcomeActivity.this.finish();
            }
        });
        buttonGoSign.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(WelcomeActivity.this, SigninActivity.class);
                startActivity(intent);
                overridePendingTransition( android.R.anim.slide_in_left,android.R.anim.slide_out_right);
                WelcomeActivity.this.finish();
            }
        });
    }
}

File: app/src/main/java/com/example/sfmapp/WelcomeActivity.kt
package com.example.sfmapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import kotlinx.android.synthetic.main.activity_welcome.*

class WelcomeActivity : AppCompatActivity() {
    var firebaseUser :FirebaseUser? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)

        buttonGoReg.setOnClickListener {
            val intent = Intent(this@WelcomeActivity,RegisterActivity::class.java)
            startActivity(intent)
            finish()
        }
        buttonGoSign.setOnClickListener {
            val intent = Intent(this@WelcomeActivity,SigninActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

    override fun onStart() {
        super.onStart()
        firebaseUser = FirebaseAuth.getInstance().currentUser
        if(firebaseUser != null){
            val intent = Intent(this@WelcomeActivity,MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }
}